pkgs/test/haskell/cabalSdist/default.nix

```diff
@@ -1,7 +1,18 @@
-{ lib, haskellPackages, runCommand }:
+{ lib, haskell, haskellPackages, runCommand }:
 
 let
-  localRaw = haskellPackages.callPackage ./local/generated.nix {};
+  src = lib.fileset.toSource {
+    root = ./local;
+    fileset = lib.fileset.unions [
+      ./local/app
+      ./local/CHANGELOG.md
+      ./local/local.cabal
+    ];
+  };
+  # This prevents the source from depending on the formatting of the ./local/generated.nix file
+  localRaw = haskell.lib.compose.overrideSrc {
+    inherit src;
+  } (haskellPackages.callPackage ./local/generated.nix {});
```

Comment:
Having the `src = ./.` line in `local/generated.nix` is not needed now, right? It can be the same as the `src` above?

Reply:
Since that's technically a generated file, I think it's cleaner to override the source from here instead

Reply:
It's named as such but is in no way generated.

Reply:
I'm pretty sure it was generated using `cabal2nix`, e.g. check see [here](https://github.com/NixOS/cabal2nix/blob/92c93710247e6a4a827515caee778f8c8b3b24ea/cabal2nix/README.md?plain=1#L44-L56), but I guess that's not very relevant at this point anyways